source/framework/vx_graph_verify.c

```diff
@@ -809,23 +809,27 @@ static vx_status ownGraphCalcInAndOutNodes(vx_graph graph)
                                         }
                                     }
                                 }
-                                else if ((vx_enum)VX_BIDIRECTIONAL == prm_next_dir)
+                                else if ((vx_enum)VX_BIDIRECTIONAL == (vx_enum)prm_next_dir)
                                 {
                                     /* Current node must be an output, proceed as if it was the only input attached */
                                     /* We must make the bidirectional parameter node the only one following the
                                        output parameter node; to do this we go through all the out nodes and replace
                                        the output parameter node by the bidirectional parameter node in the in node list */
                                     ++biputs_attached;
                                     uint32_t i;
-                                    for (i = 0; i < ownNodeGetNumOutNodes(node_cur) && ((vx_status)VX_SUCCESS == status); ++i)
+                                    for (i = 0; i < (ownNodeGetNumOutNodes(node_cur) && ((vx_status)VX_SUCCESS == status)); ++i)
```

Comment:
This placement of parenthesis doesn't look right to me, same as in previous comment.

Reply:
oh, you're totally right, thank you for reviewing, this is now fixed. I will review the other misra changes to be sure we do not have a side effect.